serialization.rst

```diff
@@ -81,16 +87,66 @@ Examples
     public $foo => 42,
     protected $prot => "wine",
     private $fpr => "cheese"
-    function bsonSerialize() : array {
+    function bsonSerialize() {
```

Comment:
AFAIK, https://wiki.php.net/rfc/return_types doesn't allow mixed return types (e.g. `array|stdClass`), nor would HHVM (I imagine I'd have to use Variant).